### PR TITLE
fix(topstories): Ctrl-click/middle mouse button does not open new tab

### LINKED ISSUE #3033: Ctrl+Click or middle mouse button click does not open the Pocket tile in a new tab
Also see https://bugzilla.mozilla.org/show_bug.cgi?id=1385871

This is because Pocket specially handles clicks to insert a referrer. We probably just need to pass in the click event details as the gBrowser helper functions should open appropriately based on event.

Probably regressed by #2959.

### PR DESCRIPTION
**Pull request by csadilek**
Fixes #3033.

### CODE REVIEW

File: system-addon/content-src/components/Card/Card.jsx [hunk 1]
```diff
@@ -31,7 +31,10 @@ class Card extends React.Component {
   }
   onLinkClick(event) {
     event.preventDefault();
-    this.props.dispatch(ac.SendToMain({type: at.OPEN_LINK, data: this.props.link}));
+    this.props.dispatch(ac.SendToMain({
+      type: at.OPEN_LINK,
+      data: Object.assign(this.props.link, {where: (event.ctrlKey || event.metaKey || event.button === 1) ? "tab" : "current"})
```
Comment (Mardak) [inline system-addon/content-src/components/Card/Card.jsx]: So this supports tabs but not the other click behaviors, e.g., "window" (on shift) and "save" (on alt) or other behaviors supported in `whereToOpenLink`:
https://searchfox.org/mozilla-central/source/browser/base/content/utilityOverlay.js#111-159

Not sure if there's a good way to just send the event over to call that method on the browser.

Comment (csadilek) [inline system-addon/content-src/components/Card/Card.jsx]: Yeah, I didn't find one :(. We have support for "window" though via the context menu (not shortcut) (incl. private windows).

I think this is better than not having it for 56, but I will leave it up to you guys. I am fine either way and can dig in more.

Comment (k88hudson) [inline system-addon/content-src/components/Card/Card.jsx]: Looks like you could call`browser.ownerGlobal.whereToOpenLink` with the event, if you can pass the whole event up

**Comment by csadilek** [comment]
OK, I can give it another try.

File: system-addon/content-src/components/Card/Card.jsx (hunk 1 above)
Comment (Mardak) [inline system-addon/content-src/components/Card/Card.jsx]: A fake event should work:
```js
const {altKey, button, ctrlKey, metaKey, shiftKey} = event;
dispatch({…
  data: … {event: {altKey, button, ctrlKey, metaKey, shiftKey}, …}
```

Comment (csadilek) [inline system-addon/content-src/components/Card/Card.jsx]: @Mardak ah I will try that. Was just looking for ways to get the event across again.

**Comment by csadilek** [comment]
@Mardak @k88hudson thanks, updated.

**Comment by as-pine-proxy** [comment]
[Treeherder: 56a25617de9c4426bcd87ec6eabe0663a41b72f3](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=56a25617de9c4426bcd87ec6eabe0663a41b72f3)

**Pull request merged**